src/types.ts

```diff
@@ -57,25 +57,42 @@ export type DidDeclineAllHandler = (opts?: {
   includingNecessary: boolean;
 }) => boolean;
 
-export interface CookieConsentHookActions {
+export interface ConsentHookActions {
   acceptCookies: (cookies: CookieTypes, options?: AcceptCookiesOptions) => void;
   declineAllCookies: () => void;
   acceptAllCookies: () => void;
   didAcceptAll: () => boolean;
   didDeclineAll: DidDeclineAllHandler;
-  cookies: CookieWrapper;
 }
 
 export type CookieConsent = CookieTypes;
 
-export interface CookieConsentHookState extends CookieConsentHookActions {
+export interface CookieConsentHookState extends ConsentHookActions {
+  cookies: CookieWrapper;
+  consent: CookieConsent;
+}
+
+export interface StorageConsentHookState extends ConsentHookActions {
+  storage: Storage;
   consent: CookieConsent;
```

Comment:
We probably also want to rename "CoocieConsent" to something like "Consent" or "ConsentState" or something similar.